### DISCUSSION TITLE
Extending "COMMON-LISP-symbols.csv" with own functions and URLs to Documentation

### DISCUSSION THREAD

**Discussion by vigh1000** [qingpeng9802/vscode-common-lisp#11] [Q&A] [answered]
Hi there, I'm posting this here because I don't think it's an "Issue" or Feature Request, but rather a question how to do it.

I wanted to expand the "COMMON-LISP-symbols.csv" with my own functions and a URL to the corresponding documentation. So that a tooltip appears and I can just click on the link. This doesn't seem to work though. Is the .csv just some kind of "output"? And is there a way to do this the correct way? 

I was able to expand `.\syntaxes\commonlisp.tmLanguage.json` with another function so it gets highlighted and add it to `\dist\web\extension.js` so the function appears in code completion. This didn't feel quite right to do though, but it works :D 

Also it really is nice to expand `.\snippets\commonlisp_snippets.json` with my own snippets!

**Comment by qingpeng9802** [comment, marked as answer]
Unfortunately, the `csv` file is only a crawling result so it is just not related to the extension app.

The processed data is at `./src/web/cl_data`. The data is just `{symbol_name: markdown_string_doc}`. The real interface of the data is at `./src/web/doc/get_doc.ts`. If you would like to add more things about tooptip, just check `./src/web/provider_interface/providers/hover_provider.ts`. 

For adding a vscode provider, see https://github.com/qingpeng9802/vscode-common-lisp/blob/master/CONTRIBUTING.md#development. If you just need it without an option, you can skip the steps and directly register the provider here `./src/web/extension.ts`.

For simple cases, modifying `commonlisp.tmLanguage.json` is the right way. Code completion is slightly complicated, but it is almost the same as hover.
